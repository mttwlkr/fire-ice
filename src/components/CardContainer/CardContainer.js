import React from 'react';

const CardContainer = () => {

  return (
    <div className='Container'>
      I'm a card container
    </div>
  )
}

export default CardContainer