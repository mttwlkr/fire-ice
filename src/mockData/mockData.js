export const mockHousesArray = [
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  },
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  }  
]

const mockHousesAction = {houses: [
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    },
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    }  
  ]
}

export const mockCardWithSwornMembers = { 
  ancestralWeapons: ["Lady Forlorn"],
  cadetBranches: [],
  coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
  currentLord: "https://www.anapioficeandfire.com/api/characters/681",
  diedOut: "",
  founded: "",
  founder: "https://www.anapioficeandfire.com/api/characters/256",
  heir: "https://www.anapioficeandfire.com/api/characters/677",
  name: "House Corbray of Heart's Home",
  overlord: "https://www.anapioficeandfire.com/api/houses/7",
  region: "",
  seats: ["Heart's Home"],
  swornMembers: [
    {url: "https://www.anapioficeandfire.com/api/characters/255", name: "Corwyn Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/256", name: "Corwyn Corbray", gender: "Male", culture: "Andal", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/447", name: "Gwayne Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/654", name: "Lucas Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/677", name: "Lyn Corbray", gender: "Male", culture: "", born: "At Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/681", name: "Lyonel Corbray", gender: "Male", culture: "", born: "In 260 AC (roughly), at Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/871", name: "Rhaena Targaryen", gender: "Female", culture: "", born: "In 116 AC, at Pentos"},
    {url: "https://www.anapioficeandfire.com/api/characters/1182", name: "Baelish", gender: "Male", culture: "Braavosi", born: ""}
  ],
  titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
  url: "https://www.anapioficeandfire.com/api/houses/81",
  words: "",
}


