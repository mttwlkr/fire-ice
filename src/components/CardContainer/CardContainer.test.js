import React from 'react';
import { CardContainer, mapStateToProps } from './CardContainer';
import { shallow } from 'enzyme';
import { mockHousesArray } from '../../mockData/mockData.js'

describe('CardContainer', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer 
      houses={mockHousesArray}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should mapStateToProps accordingly', () => {
    const mockState = {houses: ["Ive never watched Game of Thrones"]}
    const mapped = mapStateToProps(mockState)
    expect(mockState.houses).toEqual(mapped.houses)
  })
})