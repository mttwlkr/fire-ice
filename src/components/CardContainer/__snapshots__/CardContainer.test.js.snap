// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houses={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          house={
                    Object {
                              "ancestralWeapons": Array [
                                "Lady Forlorn",
                              ],
                              "cadetBranches": Array [],
                              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                              "diedOut": "",
                              "founded": "",
                              "founder": "https://www.anapioficeandfire.com/api/characters/256",
                              "heir": "https://www.anapioficeandfire.com/api/characters/677",
                              "name": "House Corbray of Heart's Home",
                              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                              "region": "",
                              "seats": Array [
                                "Heart's Home",
                              ],
                              "swornMembers": undefined,
                              "titles": undefined,
                              "url": "https://www.anapioficeandfire.com/api/houses/81",
                              "words": "",
                            }
          }
/>,
        <Card
          house={
                    Object {
                              "ancestralWeapons": Array [
                                "Lady Forlorn",
                              ],
                              "cadetBranches": Array [],
                              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                              "diedOut": "",
                              "founded": "",
                              "founder": "https://www.anapioficeandfire.com/api/characters/256",
                              "heir": "https://www.anapioficeandfire.com/api/characters/677",
                              "name": "House Corbray of Heart's Home",
                              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                              "region": "",
                              "seats": Array [
                                "Heart's Home",
                              ],
                              "swornMembers": undefined,
                              "titles": undefined,
                              "url": "https://www.anapioficeandfire.com/api/houses/81",
                              "words": "",
                            }
          }
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
            "diedOut": "",
            "founded": "",
            "founder": "https://www.anapioficeandfire.com/api/characters/256",
            "heir": "https://www.anapioficeandfire.com/api/characters/677",
            "name": "House Corbray of Heart's Home",
            "overlord": "https://www.anapioficeandfire.com/api/houses/7",
            "region": "",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": undefined,
            "titles": undefined,
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "house": Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
            "diedOut": "",
            "founded": "",
            "founder": "https://www.anapioficeandfire.com/api/characters/256",
            "heir": "https://www.anapioficeandfire.com/api/characters/677",
            "name": "House Corbray of Heart's Home",
            "overlord": "https://www.anapioficeandfire.com/api/houses/7",
            "region": "",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": undefined,
            "titles": undefined,
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Lady Forlorn",
                                    ],
                                    "cadetBranches": Array [],
                                    "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                                    "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                                    "diedOut": "",
                                    "founded": "",
                                    "founder": "https://www.anapioficeandfire.com/api/characters/256",
                                    "heir": "https://www.anapioficeandfire.com/api/characters/677",
                                    "name": "House Corbray of Heart's Home",
                                    "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                                    "region": "",
                                    "seats": Array [
                                      "Heart's Home",
                                    ],
                                    "swornMembers": undefined,
                                    "titles": undefined,
                                    "url": "https://www.anapioficeandfire.com/api/houses/81",
                                    "words": "",
                                  }
            }
/>,
          <Card
            house={
                        Object {
                                    "ancestralWeapons": Array [
                                      "Lady Forlorn",
                                    ],
                                    "cadetBranches": Array [],
                                    "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                                    "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
                                    "diedOut": "",
                                    "founded": "",
                                    "founder": "https://www.anapioficeandfire.com/api/characters/256",
                                    "heir": "https://www.anapioficeandfire.com/api/characters/677",
                                    "name": "House Corbray of Heart's Home",
                                    "overlord": "https://www.anapioficeandfire.com/api/houses/7",
                                    "region": "",
                                    "seats": Array [
                                      "Heart's Home",
                                    ],
                                    "swornMembers": undefined,
                                    "titles": undefined,
                                    "url": "https://www.anapioficeandfire.com/api/houses/81",
                                    "words": "",
                                  }
            }
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "house": Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
